tests/backup/backup_kubevirt_test.go

```diff
@@ -1335,3 +1335,217 @@ var _ = Describe("{KubevirtVMBackupRestoreWithNodeSelector}", func() {
+		//controlChannel       chan string
+		//errorGroup           *errgroup.Group
+		allVMs           []kubevirtv1.VirtualMachine
+		allVMNames       []string
+		backupName       string
+		backupNames      []string
+		freezeRuleName   string
+		unfreezeRuleName string
+		freezeRuleUid    string
+		unfreezeRuleUid  string
+	)
+
+	JustBeforeEach(func() {
+		StartPxBackupTorpedoTest("KubevirtVMWithFreezeUnfreeze", "Verify VM backup with freeze rule and without unfreeze", nil, 296422, Mkoppal, Q1FY25)
+
+		backupLocationMap = make(map[string]string)
+		providers = GetBackupProviders()
+
+		log.InfoD("scheduling applications")
+		scheduledAppContexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			taskName := fmt.Sprintf("%d-%d", 296422, i)
+			appContexts := ScheduleApplications(taskName)
+			for _, appCtx := range appContexts {
+				appCtx.ReadinessTimeout = AppReadinessTimeout
+				scheduledAppContexts = append(scheduledAppContexts, appCtx)
+			}
+		}
+	})
+
+	It("Verify VM backup with freeze rule and without unfreeze", func() {
+
+		Step("Validating applications", func() {
+			log.InfoD("Validating applications")
+			//ctx, err := backup.GetAdminCtxFromSecret()
+			//log.FailOnError(err, "Fetching px-central-admin ctx")
+			ValidateApplications(scheduledAppContexts)
+			//controlChannel, errorGroup = ValidateApplicationsStartData(scheduledAppContexts, ctx)
+		})
+
+		Step("Creating backup location and cloud setting", func() {
+			log.InfoD("Creating backup location and cloud setting")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, provider := range providers {
+				cloudCredName = fmt.Sprintf("%s-%s-%s", "cred", provider, RandomString(6))
+				backupLocationName = fmt.Sprintf("%s-%v", getGlobalBucketName(provider), RandomString(6))
+				cloudCredUID = uuid.New()
+				backupLocationUID = uuid.New()
+				backupLocationMap[backupLocationUID] = backupLocationName
+				err := CreateCloudCredential(provider, cloudCredName, cloudCredUID, BackupOrgID, ctx)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of cloud credential named [%s] for org [%s] with [%s] as provider", cloudCredName, BackupOrgID, provider))
+				err = CreateBackupLocation(provider, backupLocationName, backupLocationUID, cloudCredName, cloudCredUID, getGlobalBucketName(provider), BackupOrgID, "", true)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Creating backup location - %s", backupLocationName))
+			}
+		})
+
+		Step("Registering cluster for backup", func() {
+			log.InfoD("Registering cluster for backup")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+
+			err = CreateApplicationClusters(BackupOrgID, "", "", ctx)
+			dash.VerifyFatal(err, nil, "Creating source and destination cluster")
+
+			clusterStatus, err := Inst().Backup.GetClusterStatus(BackupOrgID, SourceClusterName, ctx)
```

Comment:
Can we add a loop here where we run the same function calls for SourceClusterName and DestinationClusterName cluster name? 

```
for clusterName in [DestinationClusterName, SourceClusterName ]{
	clusterStatus, err = Inst().Backup.GetClusterStatus(BackupOrgID, clusterName, ctx)
			log.FailOnError(err, fmt.Sprintf("Fetching [%s] cluster status", clusterName))
			dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, fmt.Sprintf("Verifying if [%s] cluster is online", clusterName))

}
```

Reply:
This is a good suggestion. Will add it. But it is there in almost all the tests.

Reply:
We need to clean such things at some point.

Reply:
Done